shared/src/dto/instructor/instructor.dto.ts

```diff
@@ -1,23 +1,151 @@
 import { ApiProperty } from '@nestjs/swagger';
-import { DtoUser } from '../user/user.dto';
+import { IsNotEmpty } from 'class-validator';
+import { DtoDriversLicenseCategory } from '../driversLicenseCategory/driversLicenseCategory.dto';
+import { DtoCreateUser, DtoUpdateUser, DtoUser } from '../user/user.dto';
 
 export class DtoInstructor {
   @ApiProperty()
+  @IsNotEmpty()
   id: number;
 
   @ApiProperty()
+  @IsNotEmpty()
   user: DtoUser;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  registrationNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  licenseNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  instructorsQualifications: number[];
+
+  @ApiProperty({ nullable: true })
+  photo: string | null | undefined;
+}
+
+export class DtoLessonInstructor {
+  @ApiProperty()
+  @IsNotEmpty()
+  id: number;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  user: DtoUser;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  registrationNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  licenseNumber: string;
+
+  @ApiProperty({
+    isArray: true,
+    type: DtoDriversLicenseCategory,
+  })
+  instructorsQualifications: DtoDriversLicenseCategory[];
+
+  @ApiProperty({ nullable: true })
+  photo: string | null | undefined;
+}
+
+export class DtoFindOneInstructor {
+  @ApiProperty()
+  user: DtoCreateUser;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  registrationNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  licenseNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  instructorsQualifications: DtoDriversLicenseCategory[];
+
+  @ApiProperty({ nullable: true })
+  photo: string | null;
+}
+
+export class DtoCreateInstructor {
+  @ApiProperty()
+  user: DtoCreateUser;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  registrationNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  licenseNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  instructorsQualifications: number[];
+
+  @ApiProperty({ nullable: true })
+  photo: string | null;
+}
+
+export class DtoUpdateInstructor {
+  @ApiProperty()
+  user: Partial<DtoUpdateUser>;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  registrationNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  licenseNumber: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  instructorsQualifications: number[];
+
+  @ApiProperty({ nullable: true })
```

Comment:
We can make this property optional by adding following properties (and to make the BE receive it as `null` when it's missing)
```suggestion
  @ApiPropertyOptional({ nullable: true, type: 'string' })
  @IsOptional()
  @IsString()
  @Transform(({ value }) => value ?? null)
```